src/troblueshooting/content.md

```diff
@@ -0,0 +1,551 @@
+1. The Pod is bound to a _Pending_ PersistentVolumeClaim
+
+Your best option is to inspect the _Events_ section in the `kubectl describe` command:
+
+```terminal|command=1|title=bash
+kubectl describe pod <pod name>
+```
+
+For error that are created as a result of ResourceQuotas, you can inspect the logs of the cluster with:
+
+```terminal|command=1|title=bash
+kubectl get events --sort-by=.metadata.creationTimestamp
+```
+
+### RunContainerError
+
+The error appears when the container is unable to start.
+
+That's even before the application inside the container starts.
+
+The issue is usually due to misconfiguration such as:
+
+- mounting a not existant volume such as ConfigMap or Secrets
+- mounting a read-only volume as read-write
+
+You should use `kubectl describe` to collect and analyse the error.
+
+## Not Ready Pods
+
+If a Pod is _Running_ but not _Ready_ it means that the Readiness probe is failing.
+
+When the Readiness probe is failing, you the Pods isn't attached to the Service, and no traffic is forwarded to that instance.
+
+A failing Readiness probe is an application-specific error, so you should inspect the _Events_ section in `kubectl describe` to identify the error.
+
+## Troubleshooting Services
+
+If you're Pods are _Running_ and _Ready_, but you're still unable to receive a response for your app, you should check if the Service is configured correctly.
+
+Services are designed to route the traffic to Pods based on their labels.
+
+So the first thing that you should check is how many Pods are target by the Service.
+
+You can do so by checking the Endpoints in the Service:
+
+```terminal|command=1|title=bash
+kubectl describe service <service-name> | grep Endpoints
+```
+
+An endpoint is a pair of `<ip address:port>`, and there should be at least one — when the Service targets (at least) a Pod.
+
+If the endpoints section is empty, there are two explanations:
+
+1. you don't have any Pod running with the correct label
+1. the label doesn't exist
+
+If you see a list of endpoints, but still can't route the traffic, then the `targetPort` in your service is the likely culprit.
+
+_How do you test the Service?_
+
+Regardless of the type of Service, you can use `kubectl-forward` to connect to it:
+
+```terminal|command=1|title=bash
+kubectl port-forward service/<service-name> 3000:80
+```
+
+Where:
+
+- `<service-name>` is the name of the Service
+- `3000` is the port that you wish to open on your computer
+- `8080` is the port exposed by the Service
+
+## Troubleshooting Ingress
+
+If you've reached this section:
+
+- the Pods are _Running_ and _Ready_
+- the Service distributes the traffic to the Pod
+
+But you still can't see a response from your cluster.
+
+At this point, the Ingress is likely misconfigured.
+
+Since the Ingress is a component that doesn't come by default in the cluster, there are different debugging techniques depending on the type of Ingress.
```

Comment:
Since the Ingress controller is a third-party component in the cluster, there are different debugging techniques depending on the type of Ingress controller.